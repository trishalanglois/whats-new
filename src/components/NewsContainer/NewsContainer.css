.news-container {
  grid-area: main;
  column-count: 2;
  column-gap: 2em;
  padding: 2em
}
