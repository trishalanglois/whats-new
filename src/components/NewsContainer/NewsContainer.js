import React from 'react';
import './NewsContainer.css'
import NewsArticle from '../NewsArticle/NewsArticle'

const NewsContainer = ({articles}) => {
  const newsArticles = articles.map((article, i) => {
    return (
      <NewsArticle
        key = {i}
        image = {article.img}
        headline = {article.headline}
        description = {article.description}
        url = {article.url}
      />
    )
  })
  return (
    <main className = 'news-container'>
      {newsArticles}
    </main>
  )
};

export default NewsContainer;
