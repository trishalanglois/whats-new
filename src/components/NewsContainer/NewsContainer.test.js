import React from 'react';
import NewsContainer from './NewsContainer';
import { shallow } from 'enzyme';

describe('NewsContainer', () => {
  it('should correctly match the snapshot with all the data passed correctly', () => {
    const articles = [
      {
        img:'b',
        headline:'b',
        description:'b',
        url:'b'
      }, {
        img:'b',
        headline:'b',
        description:'b',
        url:'b'
      }]
    const wrapper = shallow( < NewsContainer
      articles = {articles}
      />);
    expect(wrapper).toMatchSnapshot();
  })
})
