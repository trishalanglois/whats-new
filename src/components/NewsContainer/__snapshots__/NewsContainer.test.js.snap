// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should correctly match the snapshot with all the data passed correctly 1`] = `
<main
  className="news-container"
>
  <NewsArticle
    description="b"
    headline="b"
    image="b"
    key="0"
    url="b"
  />
  <NewsArticle
    description="b"
    headline="b"
    image="b"
    key="1"
    url="b"
  />
</main>
`;
